plugins/qeta-backend/src/service/router.ts

```diff
@@ -744,6 +742,52 @@ export async function createRouter({
     response.send(tags);
   });
 
+  // POST /attachments
+  router.post(
+    '/attachments',
+    multerUploaderMiddleware(config, 'image'),
+    async (request, response) => {
+      const backendBaseUrl = config.getString('backend.baseUrl');
+      const qetaUrl = `${backendBaseUrl}/api/qeta/attachments`;
+
+      const storageType =
+        config.getOptionalString('qeta.storage.type') || 'database';
+
+      const imageBuffer = await fs.promises.readFile(`${request.file?.path}`);
+
+      const attachments = await database.postAttachment(
+        storageType,
+        imageBuffer,
+      );
+
+      const attachmentID = attachments[0].id;
+      const imageURL = `${qetaUrl}/${attachmentID}`;
+
+      response.json({
+        attachmentID: attachmentID,
+        imageURL: imageURL,
+      });
+    },
+  );
+
+  // GET /attachments/:id
+  router.get('/attachments/:id', async (request, response) => {
+    const storageImagesFolderType =
+      config.getOptionalString('qeta.storage.folder') ||
+      '/tmp/backstage-qeta-images';
+
+    const { id } = request.params;
+
+    const attachment = await database.getAttachment(Number(id));
+    const imageBuffer = attachment[0].binaryImage;
+
+    const imageName = `${storageImagesFolderType}/image-${id}-${Date.now()}`;
+
+    fs.writeFileSync(imageName, imageBuffer);
+
+    return response.sendFile(imageName);
```

Comment:
Why not just return the image as attachment directly instead first writing it to fs? So something like:
```ts
res.writeHead(200, {'Content-Type': attachment.mimeType, 'Content-Length': imageBuffer.length});
res.end(imageBuffer);
```